postgres/client.sls

```diff
@@ -7,23 +7,23 @@
   {%- endif %}
 {%- endfor %}
 
-{%- if postgres.use_upstream_repo %}
-
-include:
-  - postgres.upstream
-
-{%- endif %}
-
 # Install PostgreSQL client and libraries
 
 postgresql-client-libs:
   pkg.installed:
     - pkgs: {{ pkgs }}
-{%- if postgres.use_upstream_repo %}
+  {%- if postgres.pkg_repo.name %}
+    - fromrepo: {{ postgres.pkg_repo.name }}
```

Comment:
This kind of the `fromrepo`  usage would work for YUM etc, but not for APT. Please check out the [docs](https://docs.saltstack.com/en/latest/ref/states/all/salt.states.pkgrepo.html#salt.states.pkgrepo.managed).
You need to provide distribution codename for Debian-based distros. Better to do it by exposing parameter `dist` in `codenamemap.yaml` file and use `pkg_repo.name` only as a fallback.

Reply:
Thanks, this is good feedback. I have only tested on YUM system, so good catch. I appreciate your PR reviews @vutny, they are always sharp and helpful to community!!  I'll do more testing and see if this blocks the PR.